Fix issue on headsup when a crew has vacant rider spots

tceluzza opened the pull request:
This should fix the issue we were having with open rider spots. It makes 5 separate database queries and if one comes back empty, that rider spot is left blank on the whiteboard. It also no longer shows a radio number of 0 if the spot is set as reserved or OOS. 
I also took the opportunity to add the favicon (pulled from [rpiambulance/website](https://github.com/rpiambulance/website) since Jacob mentioned it was missing.

MasterOdin commented on src/crews.js:
```diff
@@ -1,49 +1,127 @@
+
+
+const CREW_GET_D = `
+    SELECT
+        d.driver as id,
+        CONCAT(SUBSTRING(d.first_name, 1, 1), '. ', d.last_name) as name,
+        d.radionum as rn
+    FROM
+        (SELECT c.driver, m.first_name, m.last_name, m.radionum, c.date FROM crews c, members m
+            WHERE m.id = c.driver AND c.date = ?) as d
+    WHERE
+        d.date IS NOT NULL;
```
If you're going to unwind these into their own DB queries, then you can remove the subquery and get slightly better performance just using a regular query, so something like:
```
    SELECT
        c.driver AS id
        CONCAT(SUBSTRING(m.first_name, 1, 1), '. ', m.last_name) as name,
        m.radionum AS rn
    FROM
        crews AS c
    INNER JOIN members AS m ON m.id = c.driver
    WHERE c.date = ?
```

I'd also wrap this in a function so you could do:
```
function getQuery(role) {
    return `SELECT
        c.${role} AS id
        CONCAT(SUBSTRING(m.first_name, 1, 1), '. ', m.last_name) as name,
        m.radionum AS rn
    FROM
        crews AS c
    INNER JOIN members AS m ON m.id = c.${role}
    WHERE c.date = ?`;
}
```

and then call the function with `getQuery('driver')` or `getQuery('dutysup')` or etc so that way we've only got this thing written once in case want to make changes to it in the future. You could then wrap calling this in conjunction with executing the query to further reduce repetition.

tceluzza replied:
Thanks so much for the thoughts - I'll definitely get on implementing that.

ddbruce reviewed (commented):
Just one question: the query as written produces the following output due to the `<first initial. <last name>` formatting we impose:

<img width="136" alt="image" src="https://user-images.githubusercontent.com/5694673/157773550-58c1648e-1062-49d8-8e7a-723783165ad8.png">

There's no dev deployment of headsup, so I can't test if your code functions the same way or not without merging to prod.

Also, I'll take this time to say that if a spot is open, we should display something better than "OUT OF SERVICE"...it should just be blank or grayed out or something, but that's for another PR.

ddbruce commented:
Also I changed the base branch as `dev` doesn't really do anything for this repo. You'll have to reconcile some things.

tceluzza commented:
> Just one question: the query as written produces the following output due to the <first initial. <last name> formatting we impose
Yes, the new query returns the same format of name for OOS/Reserved positions:
<img width="161" alt="image" src="https://user-images.githubusercontent.com/29312937/157774557-280ffc5d-f522-41fc-b6e5-7fe88cb02b98.png">
<img width="205" alt="image" src="https://user-images.githubusercontent.com/29312937/157774600-62ab5b87-19c3-489e-9771-c6ed99761594.png">
The column name is obviously different since each position is getting queried separately, so it no longer lists it as `rider1` or `rider1rn` anymore. However, I did change it so that when it actually displays on the Headsup page, it just shows as "OUT OF SERVICE" or "RESERVED" instead of ". OUT OF SERVICE" or ". RESERVED", and a vacant rider spot is left blank. 

> You'll have to reconcile some things.
What are you referring to? Everything looks good on my end except for the codeclimate check, but I don't think there's anything I can do there.

ddbruce commented:
There's supposed to be a version check to ensure `package.json` is updated. I don't know why that's not showing right now. Also, I see this:

<img width="831" alt="image" src="https://user-images.githubusercontent.com/5694673/157777201-ed42041b-0191-4c7a-bdc7-00399cf49c8f.png">

tceluzza commented:
I *think* I've fixed both of those. I couldn't see the out-of-date warning, but I've merged the changes into my fork and updated the version in `package.json`.

ddbruce reviewed (approved):
lgtm
